src/http.zig

```diff
@@ -3133,6 +3137,14 @@ pub fn toResult(this: *HTTPClient) HTTPClientResult {
     };
 }
 
+fn updateProgress(this: *HTTPClient) void {
+    if (this.progress_node) |progress| {
+        progress.activate();
+        progress.setName(std.fmt.bufPrint(&this.progress_string_buf, "Downloading [{s:.2}/{s:.2}]", .{ bun.fmt.size(this.downloaded_bytes), bun.fmt.size(this.estimated_content_length) }) catch bun.outOfMemory());
```

Comment:
this function failing (the `catch ...`) is not actually an out of memory, but actually that the formatting requires more than the 512 bytes allotted. we can handle this more smoothly with `catch "Downloading"` to do *something* besides crashing, since that is possible.